### PR TITLE
llvm-gcc42: remove unused portgroup

### PR DESCRIPTION
**Pull request by chrstphrchvz**
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?

### CODE REVIEW

**Comment by mascguy** [comment]
It looks like blacklisting is still present, so why remove the PG...?

```
# Don't compile with self
compiler.blacklist-append macports-llvm-gcc-4.2
```

**Comment by chrstphrchvz** [comment]
> It looks like blacklisting is still present, so why remove the PG...?
> 
> 
> 
> ```
> 
> # Don't compile with self
> 
> compiler.blacklist-append macports-llvm-gcc-4.2
> 
> ```

The portgroup is needed for avoding versions of Xcode clang, it is not needed for avoiding specific MacPorts compilers.

**Review by mascguy — APPROVED** [review]
LGTM, we can merge after maintainer timeout.

**Pull request merged**
